lib/IRGen/IRGenSIL.cpp

```diff
@@ -6863,8 +6864,17 @@ void IRGenSILFunction::visitScalarPackIndexInst(ScalarPackIndexInst *i) {
 void IRGenSILFunction::visitOpenPackElementInst(swift::OpenPackElementInst *i) {
   llvm::Value *index = getLoweredSingletonExplosion(i->getIndexOperand());
 
-  // FIXME: bind the archetypes
-  (void) index;
+  llvm::SmallVector<CanPackType, 2> packs;
+  i->getOpenedGenericEnvironment()->forEachPackElementBinding(
+      [&](auto *archetype, auto *pack) { packs.push_back(CanPackType(pack)); });
+  auto packIndex = 0;
+  i->forEachDefinedLocalArchetype([&](auto archetype, auto value) {
+    auto pack = packs[packIndex];
+    auto *metadata = emitTypeMetadataPackElementRef(*this, pack, index,
+                                                    MetadataState::Complete);
+    this->bindArchetype(archetype, metadata, MetadataState::Complete, {});
+    ++packIndex;
+  });
```

Comment:
I'm not sure I see why this is being done in two loops.

Reply:
Fixed.